alicloud/service_alicloud_ecs.go

```diff
@@ -806,3 +806,62 @@ func (s *EcsService) DescribeSnapshotById(snapshotId string) (*ecs.Snapshot, err
 	}
 	return &resp.Snapshots.Snapshot[0], nil
 }
+
+func (s *EcsService) DescribeSnapshotPolicy(snapshotId string) (*ecs.AutoSnapshotPolicy, error) {
+	args := ecs.CreateDescribeAutoSnapshotPolicyExRequest()
+	args.AutoSnapshotPolicyId = snapshotId
+
+	raw, err := s.client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+		return ecsClient.DescribeAutoSnapshotPolicyEx(args)
+	})
+	if err != nil {
+		return nil, WrapErrorf(err, DefaultErrorMsg, args.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	resp := raw.(*ecs.DescribeAutoSnapshotPolicyExResponse)
+	if resp == nil || len(resp.AutoSnapshotPolicies.AutoSnapshotPolicy) < 1 {
+		return nil, WrapErrorf(err, NotFoundMsg, args.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	return &resp.AutoSnapshotPolicies.AutoSnapshotPolicy[0], nil
+}
+
+func (s *EcsService) DescribeDisksBySnapshotPolicyId(id string) (disk []ecs.Disk, err error) {
+	req := ecs.CreateDescribeDisksRequest()
+	req.AutoSnapshotPolicyId = id
+	raw, err := s.client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+		return ecsClient.DescribeDisks(req)
+	})
+	if err != nil {
+		err = WrapErrorf(err, DefaultErrorMsg, req.GetActionName(), AlibabaCloudSdkGoERROR)
+		return
+	}
+	resp, _ := raw.(*ecs.DescribeDisksResponse)
+	if resp == nil || len(resp.Disks.Disk) < 1 {
+		err = WrapErrorf(GetNotFoundErrorFromString(GetNotFoundMessage("ECS Disk with AutoSnapshotPolicy", id)), NotFoundMsg, req.GetActionName(), AlibabaCloudSdkGoERROR)
+		return
+	}
+	return resp.Disks.Disk, nil
+}
+
+func (s *EcsService) WaitForSnapshotPolicy(snapshotPolicyId string, status Status, timeout int) error {
+	if timeout <= 0 {
+		timeout = DefaultTimeout
+	}
+	deadLine := time.Now().Add(time.Duration(timeout) * time.Second)
+	for {
+		snapshotPolicy, err := s.DescribeSnapshotPolicy(snapshotPolicyId)
+		if err != nil {
+			return WrapError(err)
+		}
+
+		if snapshotPolicy.Status == string(status) {
+			return nil
+		}
+
+		if time.Now().After(deadLine) {
+			return WrapErrorf(GetTimeErrorFromString("ECS snapshot policy"), DefaultTimeoutMsg, "WaitForSnapshotPolicy", ProviderERROR)
```

Comment:
THe line 863 should be `return WrapErrorf(err, WaitTimeoutMsg, id, GetFunc(1), timeout, snapshotPolicy.Status, string(status), ProviderERROR)
`